shepherd_bms/src/eepromdirectory.cpp

```diff
@@ -0,0 +1,82 @@
+#include "eepromdirectory.h"
+
+void eepromInit()
+{   
+    // Initialize EEPROM addresses given data and length
+
+    int i = 1;
+    int offset = 0;
+
+    /* initialize root offset to zero */
+    eeprom_data[0].offset = EEPROM_ROOT_ADDR;
+
+    /* continue through array, setting offsets */
+    while (eeprom_data[i].id != NULL)
+    {
+        offset += eeprom_data[i-1].size;
+        eeprom_data[i].offset = offset;
+        i++;
+    }
+}
+
+int eepromGetIndex(char *key)
+{
+    /* find the index of the key in the eeprom_data array */
+    int i = 0;
+    while (eeprom_data[i].id != NULL)
+    {
+        if (eeprom_data[i].id == key)
+        {
+            return eeprom_data[i].offset;
+        }
+
+        i++;
+    }
+    return -1;
+}
+
+void logFault(uint32_t fault_code)
```

Comment:
Probably want a ```getRecentFault``` function too that returns the most recent fault, or returns an array of the faults in order of most recent to least recent

Reply:
im realizing naming and comments were not clear, so ill update those, but this goes for this comment and next: "getRecentFault" function is essentially implemented in updateFaults. updateFaults fills a static, local/volatile array with the current data in eeprom. I weighed using a static array with a function that just updates the data vs a function that returns a pointer to the data, but ended up with this way. So basically the user just needs to access that local array ad itll have the most up to data set of faults as long as "updateFaults" has been called

Reply:
i felt like this way creates a certain level of consistency with how we manage bms data, since anaylzer is one set of data that just reupdates with each loop